cmd/config-migrate.go

```diff
@@ -27,6 +28,39 @@ import (
 // DO NOT EDIT following message template, please open a github issue to discuss instead.
 var configMigrateMSGTemplate = "Configuration file %s migrated from version '%s' to '%s' successfully.\n"
 
+var errConfigOld = errors.New("configuration version is older than expected")
+
+// Check if the config version is the latest,
+// returns nil if no migration needed, otherwise
+// proceed to migrate.
+func checkConfigForMigrate() error {
+	configFile := getConfigFile()
+
+	type serverConfigVersion struct {
+		Version string
+	}
+
+	srvCfgVersion := &serverConfigVersion{}
+	_, err := quick.Load(configFile, srvCfgVersion)
+	if os.IsNotExist(err) {
+		// Shouldn't happen, proceed and fail while
+		// reading actual config file.
+		return nil
+	} else if err != nil {
+		// On any other file read errors, fail.
+		return err
+	}
+
+	// Config version of expected version, no need to migrate.
+	if srvCfgVersion.Version == v18 {
+		return nil
+	}
+
+	// Config is readable but older, proceed to migrate.
+	return errConfigOld
+}
+
+// Migrates all config versions from "1" to "18".
 func migrateConfig() error {
```

Comment:
This patch addresses only to the latest version. However the problem is still there in migration because of we do 1 to N version migration irrespective of current version of the config.  For example,
```go
type ConfigV6 struct {
  Version string
  Value string
}

type ConfigV7 struct {
  Version string
  Value []string
}
```
When config file is not current version and which is in version 7 like
```json
{
     "Version" : "7",
    "Value": ["myval1",  "myval2"]
}
```

the current logic still breaks
```go
	// Migrate version '6' to '7'.
	if err := migrateV6ToV7(); err != nil {
		return err
	}
```